### PR TITLE
add truecolor support for tmux (from upstream master)

### PR DESCRIPTION
**Pull request by therealpxc**
This won't be necessary with tmux-2.2, but that release may be quite a while yet. The patch is taken from here: https://github.com/tmux/tmux/issues/112

### CODE REVIEW

File: pkgs/tools/misc/tmux/default.nix [hunk 1]
```diff
@@ -13,6 +13,12 @@ stdenv.mkDerivation rec {
 
   buildInputs = [ ncurses libevent ];
 
+#  prePatch = ''
+#    sed -e '/256colors.pl/d;d;d' -i 112.diff
+#  '';
```
Comment (globin) [inline pkgs/tools/misc/tmux/default.nix]: Is this intentional?

File: pkgs/tools/misc/tmux/default.nix [hunk 2]
```diff
@@ -13,6 +13,12 @@ stdenv.mkDerivation rec {
...
+
+  patches = copyPathToStore ./tmux-truecolor-gh112.patch;
```
Comment (globin) [inline pkgs/tools/misc/tmux/default.nix]: The patch does not contain anything

**Comment by therealpxc** [comment]
I am a dummy! I've replaced the empty patch file with the right one and cleaned up the unnecessary commented-out bits in tmux/default.nix.

Thanks for looking this over, and sorry I made such obvious boo-boos.

**Comment by therealpxc** [comment]
Is this now acceptable?

**Comment by jgillich** [comment]
Why exactly does this belong in nixpkgs? It's adding a feature to a package, not something we usually do. You can always override the package.

**Comment by therealpxc** [comment]
It may not really belong in nixpkgs. It was a sort of first attempt at contributing. If we want to hew closer to upstream, that's fine.

**Comment by assaf758** [comment]
While trying 24bit-color support, I noticed there is some issue (when comparing to non-tmux terminal) - see http://ibin.co/2YAmCLcoBNRC.
Happens with two tests (see below) and two terminals (termite and gnome-terminal):
https://raw.githubusercontent.com/robertknight/konsole/master/tests/color-spaces.pl
https://github.com/gnachman/iTerm2/blob/master/tests/24-bit-color.sh

**Pull request closed without merging**